crates/duty-tracker/src/contract_manager.rs

```diff
@@ -261,12 +245,15 @@ impl ContractManager {
                         info!(%next, "successfully rpc sync'ed block");
                         for duty in duties {
                             info!(%duty, "starting duty execution from lagging blocks");
-                            if let Err(e) =
-                                execute_duty(ctx.cfg.clone(), output_handles.clone(), duty.clone())
-                                    .await
-                            {
-                                error!(%e, "failed to execute duty")
-                            };
+                            let cfg = cfg.clone();
+                            let output_handles = output_handles.clone();
+                            task::spawn(async move {
+                                if let Err(e) =
+                                    execute_duty(cfg, output_handles, duty.clone()).await
+                                {
+                                    error!(%e, %duty, "failed to execute duty");
+                                }
+                            });
```

Comment:
Yeah we can't do this. This will reintroduce the concurrent deposit bug. Remove the task::spawn from here.

Reply:
Done in 8e258e4876241d947fab9769b2844657e9be92cc